EasyPost.Tests/ClientTest.cs

```diff
@@ -109,6 +111,51 @@ public void TestHttpClientOverride()
             Assert.Equal(httpClient, overrideClient.CustomHttpClient);
         }
 
+        [Fact]
+        public async void TestHttpClientCustomProxy()
+        {
+            const string proxyAddress = "localhost:8888";
+
+            // Define a custom proxy in a custom HttpClientHandler in a custom HttpClient
+            HttpClientHandler handler = new()
+            {
+                UseProxy = true,
+                Proxy = new WebProxy($"http://{proxyAddress}"),
+            };
+            HttpClient httpClient = new(handler: handler);
+
+            Client client = new(new ClientConfiguration(FakeApikey)
+            {
+                CustomHttpClient = httpClient,
+            });
+
+            Assert.Equal(httpClient, client.CustomHttpClient);
+
+            // Assert that the proxy is set in the HttpClient by attempting to make a request (should fail due to invalid proxy address)
+            try
+            {
+                await client.Address.Create(new Parameters.Address.Create());
```

Comment:
Can we have a cassette for this unit test so it doesn't make a live API call?

Reply:
It doesn't end up making a call, because the failure to connect to the proxy happens way before it ever reached the API. I tried recording a cassette for it, but it doesn't even establish a connection, so there's nothing to record.